File: backend/app/schemas/levels.py
"""Schemas para /system/levels + /system/level-permissions."""
from pydantic import BaseModel


class LevelOut(BaseModel):
    level: int
    label: str
    description: str
    is_reserved: bool
    permissions: list[str]  # permisos actuales asignados al nivel

    model_config = {"from_attributes": True}


class LevelUpdate(BaseModel):
    label: str | None = None
    description: str | None = None
    is_reserved: bool | None = None


class LevelsPayload(BaseModel):
    """Vista completa para el frontend (tab Niveles de usuarios)."""
    levels: list[LevelOut]
    permissions: list[str]
    restricted: list[str]


class MatrixUpdate(BaseModel):
    """PATCH bulk para la matriz: {level: [permissions]}."""
    matrix: dict[int, list[str]]

File: frontend/src/lib/api/levels.ts
// API: niveles + matriz de permisos (solo level 9).
import { request } from './client'

export interface LevelDetail {
  level: number
  label: string
  description: string
  is_reserved: boolean
  permissions: string[]
}

export interface LevelsPayload {
  levels: LevelDetail[]
  permissions: string[]
  restricted: string[]
}

export const levelsApi = {
  list: () => request<LevelsPayload>('/system/levels'),
  updateMeta: (level: number, body: { label?: string; description?: string }) =>
    request<LevelDetail>(`/system/levels/${level}`, {
      method: 'PATCH',
      body: JSON.stringify(body),
    }),
  updateMatrix: (matrix: Record<number, string[]>) =>
    request<LevelsPayload>('/system/levels/matrix', {
      method: 'PATCH',
      body: JSON.stringify({ matrix }),
    }),
}
